modules/measure/src/main/java/com/opengamma/strata/measure/ValuationZoneTimeDefinition.java

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.measure;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+import org.joda.beans.impl.direct.DirectPrivateBeanBuilder;
+
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.data.scenario.MarketDataBox;
+import com.opengamma.strata.data.scenario.ScenarioArray;
+
+/**
+ * Definition of valuation zone and time. 
+ * <p>
+ * This contains {@code ZoneId} and a set of {@code LocalTime} to create {@code ZonedDateTime} from {@code LocalDate}.
+ */
+@BeanDefinition(builderScope = "private")
+public final class ValuationZoneTimeDefinition
+    implements ImmutableBean, Serializable {
+
+  /**
+   * The local time.
+   * <p>
+   * The local time in {@code zoneId}. 
+   * The number of the scenarios must be coherent to that of {@code ScenarioMarketData}.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ScenarioArray<LocalTime> localTimes;
+  /**
+   * The zone ID.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ZoneId zoneId;
+
+  /**
+   * Obtains an instance.
+   * 
+   * @param localTimes  the local time
+   * @param zoneId  the zone ID
+   * @return the instance
+   */
+  public static ValuationZoneTimeDefinition of(ScenarioArray<LocalTime> localTimes, ZoneId zoneId) {
```

Comment:
Could add a factory that takes `(ZoneId, LocalTime...)` which would mean that the zone should come first here too.